src/core/modules/StrategyManager.sol

```diff
@@ -0,0 +1,686 @@
+            if (msg.sender != bApp) revert InvalidBAppOwner(msg.sender, bApp);
+            receiver = bApp;
+            _exitStrategy(strategyId, bApp, token);
+        }
+
+        strategyTokenShares.totalTokenBalance -= amount;
+        s.slashingFund[receiver][token] += amount;
+
+        if (strategyTokenShares.totalTokenBalance == 0) {
+            delete s.strategyTokenShares[strategyId][token].totalTokenBalance;
+            delete s.strategyTokenShares[strategyId][token].totalShareBalance;
+            s.strategyTokenShares[strategyId][token].currentGeneration += 1;
+        }
+
+        emit IStrategyManager.StrategySlashed(strategyId, bApp, token, amount, receiver);
+    }
+
+    function _exitStrategy(uint32 strategyId, address bApp, address token) private {
+        CoreStorageLib.Data storage s = CoreStorageLib.load();
+        s.obligations[strategyId][bApp][token].percentage = 0;
+
+        emit IStrategyManager.ObligationUpdated(strategyId, bApp, token, 0);
+    }
+
+    function _adjustObligation(uint32 strategyId, address bApp, address token, uint256 amount) internal {
+        CoreStorageLib.Data storage s = CoreStorageLib.load();
+        ICore.Obligation storage obligation = s.obligations[strategyId][bApp][token];
+        ICore.Shares storage strategyTokenShares = s.strategyTokenShares[strategyId][token];
+        uint256 currentStrategyBalance = strategyTokenShares.totalTokenBalance;
+        uint256 currentObligatedBalance = obligation.percentage * currentStrategyBalance / MAX_PERCENTAGE;
+        uint256 postSlashStrategyBalance = currentStrategyBalance - amount;
+        uint256 postSlashObligatedBalance = currentObligatedBalance - amount;
+        if (postSlashStrategyBalance == 0) {
+            s.obligations[strategyId][bApp][token].percentage = 0;
+            emit IStrategyManager.ObligationUpdated(strategyId, bApp, token, 0);
+        } else {
+            uint32 postSlashPercentage = uint32(postSlashObligatedBalance * MAX_PERCENTAGE / currentStrategyBalance);
+            s.obligations[strategyId][bApp][token].percentage = postSlashPercentage;
+            emit IStrategyManager.ObligationUpdated(strategyId, bApp, token, postSlashPercentage);
+        }
+    }
+
+    /// @notice Withdraw the slashing fund for a token
+    /// @param token The address of the token
+    /// @param amount The amount to withdraw
+    function withdrawSlashingFund(address token, uint256 amount) external nonReentrant {
+        if (token == ETH_ADDRESS) revert InvalidToken();
+
+        _withdrawSlashingFund(token, amount);
+
+        IERC20(token).safeTransfer(msg.sender, amount);
+
+        emit IStrategyManager.SlashingFundWithdrawn(token, amount);
+    }
+
+    /// @notice Withdraw the slashing fund for ETH
+    /// @param amount The amount to withdraw
+    function withdrawETHSlashingFund(uint256 amount) external nonReentrant {
+        _withdrawSlashingFund(ETH_ADDRESS, amount);
+
+        payable(msg.sender).transfer(amount);
```

Comment:
I suggest to use `call` here. If msg.sender is a contract with logic in `receive` or `fallback` functions, the transfer will be reverted because of gas limit reached.
```
(bool success, ) = payable(msg.sender).call{value: amount}("");
if(!success) revert WithdrawTransferFailed();
```